.github/workflows/publish-docs.yml

```diff
@@ -0,0 +1,63 @@
+name: Publish Docs on GitHub Pages
+
+on:
+  pull_request:
```

Comment:
TODO

Reply:
I think it's better to trigger building flow on pull requests and deploy new docs version only on merge.